File: app/src/main/java/com/riders/thelab/navigator/Navigator.java
package com.riders.thelab.navigator;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;

import com.riders.thelab.ui.contacts.addcontact.AddContactActivity;
import com.riders.thelab.ui.mainactivity.MainActivity;
import com.riders.thelab.ui.splashscreen.SplashScreenActivity;

import javax.inject.Inject;

import timber.log.Timber;

public class Navigator {

    private final Activity context;

    @Inject
    public Navigator(Activity context) {
        this.context = context;
    }


    /* Activities */
    public void callSplashActivity() {
        context.startActivity(new Intent(context, SplashScreenActivity.class));
    }

    public void callMainActivity() {
        context.startActivity(new Intent(context, MainActivity.class));
    }

    /* Detail Activities */
    public void callContactDetailActivity(Intent intent) {
        context.startActivity(intent);
    }

    public void callAddContactActivity() {
        context.startActivity(new Intent(context, AddContactActivity.class));
    }

    public void callMultipaneDetailActivity(Intent intent) {
        context.startActivity(intent);
    }

    public void callYoutubeDetailActivity(Intent intent, @Nullable Bundle optionsCompat, boolean isWithTransition) {
        if (isWithTransition && null != optionsCompat) {
            Timber.d("Apply activity transition");
            context.startActivity(intent, optionsCompat);
        } else {
            Timber.d("Swap without transition");
            context.startActivity(intent);
        }
    }

    /* Activities */
    public void callIntentActivity(Class<? extends Activity> targetClass) {
        context.startActivity(new Intent(context, targetClass));
    }

    /* Packages */
    public void callIntentForPackageActivity(String intentPackageName) {
        context.startActivity(
                context
                        .getPackageManager()
                        .getLaunchIntentForPackage(intentPackageName));
    }
}

File: app/src/main/java/com/riders/thelab/navigator/Navigator.kt
package com.riders.thelab.navigator

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import com.riders.thelab.feature.settings.main.SettingsActivity
import com.riders.thelab.feature.weather.ui.WeatherActivity
import com.riders.thelab.ui.contacts.addcontact.AddContactActivity
import com.riders.thelab.ui.login.LoginActivity
import com.riders.thelab.ui.mainactivity.MainActivity
import com.riders.thelab.ui.signup.SignUpActivity
import com.riders.thelab.ui.splashscreen.SplashScreenActivity
import timber.log.Timber

class Navigator(private val activity: Activity) {

    /* Activities */
    fun callSplashActivity() {
        activity.startActivity(Intent(activity, SplashScreenActivity::class.java))
    }

    fun callLoginActivity() {
        activity.startActivity(Intent(activity, LoginActivity::class.java))
    }

    fun callLoginActivity(intent: Intent, optionsCompat: Bundle) {
        Timber.d("Apply login activity transition")
        activity.startActivity(intent, optionsCompat)
    }

    fun callSignUpActivity() {
        activity.startActivity(Intent(activity, SignUpActivity::class.java))
    }

    fun callSignUpActivity(intent: Intent, optionsCompat: Bundle) {
        Timber.d("Apply login activity transition")
        activity.startActivity(intent, optionsCompat)
    }

    fun callMainActivity() {
        activity.startActivity(Intent(activity, MainActivity::class.java))
    }

    fun callSettingsActivity() {
        activity.startActivity(
            Intent(
                activity,
                SettingsActivity::class.java
            )
        )
    }

    /* Detail Activities */
    fun callContactDetailActivity(intent: Intent) {
        activity.startActivity(intent)
    }

    fun callAddContactActivity() {
        activity.startActivity(Intent(activity, AddContactActivity::class.java))
    }

    fun callMultipaneDetailActivity(intent: Intent) {
        activity.startActivity(intent)
    }

    fun callYoutubeDetailActivity(intent: Intent, optionsCompat: Bundle) {
        Timber.d("Apply activity transition")
        activity.startActivity(intent, optionsCompat)
    }

    fun callWeatherActivity() {
        activity.startActivity(Intent(activity, WeatherActivity::class.java))
    }

    /* Activities */
    fun callIntentActivity(targetClass: Class<out Activity?>?) {
        if (null == targetClass) {
            Timber.e("target class is null")
            return
        } else {
            activity.startActivity(Intent(activity, targetClass))
        }
    }

    /* Packages */
    fun callIntentForPackageActivity(intentPackageName: String) {
        activity.startActivity(activity.packageManager.getLaunchIntentForPackage(intentPackageName))
    }
}